### ISSUE TITLE
cant run IOS-sim with: $GOPATH/bin/qtdeploy run ios-simulator

### ISSUE THREAD

**Issue by joeblew99** [therecipe/qt#236]
Really nice how this is in the cli now.
https://github.com/therecipe/qt/blob/master/internal/cmd/deploy/deploy.go#L987

```
$GOPATH/bin/qtdeploy run ios-simulator
ERRO[0004] failed to run command                         cmd="xcrun instruments -w iPhone 7 Plus (10.2)#" env= error="exit status 255" func=RunCmdOptional name="start simulator for ios-simulator on darwin"
Waiting for device to boot...
Instruments Usage Error: No template (-t) specified
instruments, version 8.2 (61756)
usage: instruments [-t template] [-D document] [-l timeLimit] [-i #] [-w device] [[-p pid] | [application [-e variable value] [argument ...]]]

```

i think i need to setup a sim device first ?

SO i found an awesome golang package that manages ios simulators, with test directors and all that good stuff. It would solve lots of problems with testing MANY IOS devices in one hit.

https://github.com/jwfriese/iossimulator/blob/master/simctlwrapper.go

here is a project that uses it:
https://github.com/jwfriese/OMGFruit/blob/master/script/test.go

For now, its more about run getting "$GOPATH/bin/qtdeploy run ios-simulator" to run on everyones osx hardware with no cliches. I dont have too much experience with this, but it might be a matter of parsing what they have installed and then asking them what device to use ?

**Comment by joeblew99** [comment]
Actually the standard code in the repo does work, and it did open the simulator.
I just had to run it twice for some reason.

Still i hope the link i found for managing devices via golang is useful and we think of a way to integrate it that works for all. 

Will leave this open for now ? Let me know...

**Comment by therecipe** [comment]
>I dont have too much experience with this, but it might be a matter of parsing what they have installed and then asking them what device to use ?

Yes, that would probably be the best option, as the current hardcoded `iPhone 7 Plus (10.2)` is only available in Xcode 8.2. 

>Actually the standard code in the repo does work, and it did open the simulator.
I just had to run it twice for some reason.

Yeah, it takes some time 10-15 min or so, when you start a new simulator for the first time.

>Still i hope the link i found for managing devices via golang is useful and we think of a way to integrate it that works for all.

Yes, it's useful, for example I didn't figured out yet how to check if the device is properly started up.
And `PrintSpringboardServiceAvailability` seems to do that.

So, yeah leave it open :)
